### PR TITLE
Riley/c 8312 make inbox20 designs use new inbox20

### PR DESCRIPTION
**Pull request by rileylnapier**
## Description

switch useInbox hook to use the new apis for fetching messages and tracking events

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> `Fix [#1]()`

### CODE REVIEW

File: packages/react-hooks/src/inbox/reducer.ts [hunk 1]
```diff
@@ -62,24 +61,35 @@ import {
...
   opened: message.opened,
   read: lastMarkedAllRead
     ? lastMarkedAllRead > new Date(message.created).getTime()
       ? true
       : message.read
     : message.read,
-  title: message.content.title,
-  trackingIds: message.content.trackingIds,
+  title: message.title,
+  //trackingIds: message.content.trackingIds,
```
Comment (suhasdeshpande) [inline packages/react-hooks/src/inbox/reducer.ts:92]: need this?

File: packages/react-hooks/src/inbox/use-inbox-actions.ts [hunk 2]
```diff
@@ -60,37 +53,66 @@ interface IInboxActions {
...
+      from?: number;
+      status?: "read" | "unread";
+    } = {
+      from,
+      status:
+        tab.filters.isRead === false
+          ? "unread"
+          : tab.filters.isRead === true
+          ? "read"
+          : undefined,
```
Comment (suhasdeshpande) [inline packages/react-hooks/src/inbox/use-inbox-actions.ts:68]: what is this case?

File: packages/react-inbox/src/components/Messages2.0/Message.tsx [hunk 3]
```diff
@@ -184,6 +187,14 @@ const MessageWrapper: React.FunctionComponent<
     }
   };
 
+  useOnScreen(messageHoverRef, () => {
+    if (opened || !messageId) {
+      return;
+    }
+
+    markMessageOpened(messageId);
```
Comment (suhasdeshpande) [inline packages/react-inbox/src/components/Messages2.0/Message.tsx:195]: how do we handle failures or error state here?

File: packages/react-inbox/src/components/Messages2.0/actions/index.tsx [hunk 4]
```diff
@@ -78,32 +77,23 @@ const MessageActions: React.FunctionComponent<{
 
   const handleEvent = (eventName: string) => (event: React.MouseEvent) => {
     event?.preventDefault();
+    if (!messageId) {
```
Comment (suhasdeshpande) [inline packages/react-inbox/src/components/Messages2.0/actions/index.tsx:80]: what are situations where we won't have messageId?

**Review by suhasdeshpande — APPROVED** [review]

File: packages/react-hooks/src/inbox/reducer.ts (hunk 1 above)
Comment (rileylnapier) [reply packages/react-hooks/src/inbox/reducer.ts:92]: im not sure exactly tbh, the new inbox apis have all the tracking ids in the db so we don't need to pass them back

File: packages/react-hooks/src/inbox/use-inbox-actions.ts (hunk 2 above)
Comment (rileylnapier) [reply packages/react-hooks/src/inbox/use-inbox-actions.ts:68]: the filter previously was "isRead": true | false and now its "status": "read" | "unread" so making the mapping between the two kinda sucks

File: packages/react-inbox/src/components/Messages2.0/Message.tsx (hunk 3 above)
Comment (rileylnapier) [reply packages/react-inbox/src/components/Messages2.0/Message.tsx:195]: we don't really handle api failures at all.  i wonder what we could do better here

File: packages/react-inbox/src/components/Messages2.0/actions/index.tsx (hunk 4 above)
Comment (rileylnapier) [reply packages/react-inbox/src/components/Messages2.0/actions/index.tsx:80]: i think we should always have it, let me verify

File: packages/react-hooks/src/inbox/use-inbox-actions.ts [hunk 5]
```diff
@@ -162,28 +189,25 @@ const useInboxActions = (): IInboxActions => {
       dispatch(setCurrentTab(newTab));
     },
     fetchMessages: (payload?: IFetchMessagesParams) => {
+      const searchParams: IGetInboxMessagesParams = {
```
Comment (rileylnapier) [inline packages/react-hooks/src/inbox/use-inbox-actions.ts:192]: make sure this handles status

File: packages/react-hooks/src/inbox/use-inbox-actions.ts [hunk 6]
```diff
@@ -45,13 +41,10 @@ interface IInboxActions {
   getUnreadMessageCount: (params?: IGetMessagesParams) => void;
   init: (inbox: IInbox) => void;
   markAllAsRead: () => void;
-  markMessageArchived: (
```
Comment (rileylnapier) [inline packages/react-hooks/src/inbox/use-inbox-actions.ts:48]: make this not breaking change

**Review by suhasdeshpande — APPROVED** [review]

**Pull request merged**